pkg/streamingpromql/operators/binary_operation.go

```diff
@@ -449,9 +508,13 @@ func (b *BinaryOperation) mergeOneSide(data []types.InstantVectorSeriesData, sou
 		}
 	}
 
+	// Re-slice the ds.data with just the series with floats to make the rest of our job easier
+	// Because we aren't re-sorting here it doesn't matter that ds.sourceSeriesIndices remains longer.
+	data := ds.data[:remainingSeriesWithFloats]
```

Comment:
We should return the slices for the empty series to the pool (they might have length 0, but they may have non-zero capacity, so we can reuse them).